packages/application-config/src/load-config.ts

```diff
@@ -1,83 +1,86 @@
 import type { JSONSchemaForCustomViewConfigurationFiles } from './schemas/generated/custom-view.schema';
 
-function doesFileExist(path: string): boolean {
+// See the following issues for more context, contributing to failing Jest tests:
+//  - Issue: https://github.com/nodejs/node/issues/40058
+//  - Resolution: https://github.com/nodejs/node/pull/48510 (Node v20.8.0)
+// Copied from @commitlint/load/src/utils/load-config.ts
+export const isDynamicAwaitSupported = () => {
+  const [major, minor] = process.version
+    .replace('v', '')
+    .split('.')
+    .map((val) => parseInt(val));
+
+  return major >= 20 && minor >= 8;
+};
+
+// Is the given directory set up to use ESM (ECMAScript Modules)?
+// Copied from @commitlint/load/src/utils/load-config.ts
+export const isEsmModule = async (cwd: string) => {
+  const packagePath = path.join(cwd, 'package.json');
+
   try {
-    fs.accessSync(path);
-    return true;
+    await fs.promises.access(packagePath, fs.constants.F_OK);
+    const packageJSON = await fs.promises.readFile(packagePath, {
+      encoding: 'utf-8',
+    });
+    return JSON.parse(packageJSON)?.type === 'module';
   } catch (error) {
     return false;
   }
-}
-// Helper function to find the package root path from the current location,
-// for instance in respect to both source files and dist files.
-const findPackageRootPath = (dir: string): string => {
-  const packageJsonPath = path.join(dir, 'package.json');
-  if (doesFileExist(packageJsonPath)) {
-    return dir;
-  }
-  const parentDir = path.join(dir, '..');
-  return findPackageRootPath(parentDir);
 };
 
-const loadJsModule: LoaderSync = (filePath) => {
-  const packageRootPath = findPackageRootPath(
-    // Start from the parent folder
-    path.join(__dirname, '..')
-  );
-  // Load the JS module using a child process. This is primarly to avoid
-  // unwanted behaviors using `@babel/register` in the main process.
-  // The loader script does the actual `require` of the given `filePath`
-  // and uses `@babel/register` to correctly parse and execute the file.
-  // The "required module output" is then written into `stdout` and parsed
-  // as JSON.
-  const output = execFileSync(
-    'node',
-    [path.join(packageRootPath, 'scripts/load-js-module.js'), filePath],
-    { encoding: 'utf8' }
-  );
-  return JSON.parse(output);
-};
+const createExplorerFor = async (configFileName: string) => {
+  // Copied from @commitlint/load/src/utils/load-config.ts
+  let tsLoaderInstance: Loader | undefined;
+  const tsLoader: Loader = (...args) => {
+    if (!tsLoaderInstance) {
+      tsLoaderInstance = TypeScriptLoader();
+    }
+    return tsLoaderInstance(...args);
+  };
 
-const createExplorerFor = (configFileName: string) => {
-  return cosmiconfigSync(configFileName, {
-    // Restrict the supported file formats / names
+  // If dynamic await is supported (Node >= v20.8.0) or directory uses ESM, support
```

Comment:
**nit**: I still struggle to understand why we need to use async loaders and how `defaultLoaders` seems to change its behaviour across node versions (we were already using it but not awaiting the search files).

Reply:
I just copied it from here: https://github.com/conventional-changelog/commitlint/blob/fed05e2ab12792ce488f34a580643669ab6ac831/%40commitlint/load/src/utils/load-config.ts#L33-L38

In any case, it refers to this part of the docs: https://github.com/cosmiconfig/cosmiconfig/blob/main/README.md#loading-js-modules
<img width="1038" alt="image" src="https://github.com/user-attachments/assets/e60766e2-1cad-41aa-8904-fdf1a0b2946c">